### PR TITLE
Document NetworkEncryptionUtils

### PR DESCRIPTION
**Pull request by IotaBread**
Mostly a port from yarn

### CODE REVIEW

**Review by OroArmor — APPROVED** [review]

File: mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping [hunk 1]
```diff
@@ -15,53 +32,158 @@ CLASS net/minecraft/unmapped/C_mgrgevhd net/minecraft/network/encryption/Network
...
+		COMMENT @see #decodeRsaPrivateKeyPem(String)
+		COMMENT @see #decodeEncodedRsaPublicKey(byte[])
 		ARG 0 key
 	METHOD m_diyouuze encodeRsaPrivateKey (Ljava/security/PrivateKey;)Ljava/lang/String;
-		ARG 0 privateKey
+		COMMENT Encodes an RSA private {@code key} to a PEM-formatted key string.
+		COMMENT
+		COMMENT <p>
+		COMMENT A PEM-formatted RSA private key is {@value #RSA_PRIVATE_KEY_HEADER}, followed
+		COMMENT by Base64 encoded PCKS #8 encoded key, followed by {@value #RSA_PRIVATE_KEY_FOOTER}.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping]:
```suggestion
		COMMENT by Base64 encoded PCKS #8-encoded key, followed by {@value #RSA_PRIVATE_KEY_FOOTER}.
```
for consistency

File: mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping [hunk 2]
```diff
@@ -15,53 +32,158 @@ CLASS net/minecraft/unmapped/C_mgrgevhd net/minecraft/network/encryption/Network
...
+		COMMENT @see #decodeRsaPublicKeyPem(String)
+		ARG 0 key
 	METHOD m_rebtwskh (Ljava/lang/String;)Lcom/mojang/serialization/DataResult;
 		ARG 0 keyPem
 	METHOD m_tnwvnyev decodeRsaPrivateKeyPem (Ljava/lang/String;)Ljava/security/PrivateKey;
+		COMMENT Decodes a PEM-formatted RSA private key.
+		COMMENT
+		COMMENT <p>
+		COMMENT A PEM-formatted RSA private key is {@value #RSA_PRIVATE_KEY_HEADER}, followed
+		COMMENT by Base64 encoded PCKS #8 encoded key, followed by {@value #RSA_PRIVATE_KEY_FOOTER}.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping]:
```suggestion
		COMMENT by Base64 encoded PCKS #8-encoded key, followed by {@value #RSA_PRIVATE_KEY_FOOTER}.
```
consistency

File: mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping [hunk 3]
```diff
@@ -15,53 +32,158 @@ CLASS net/minecraft/unmapped/C_mgrgevhd net/minecraft/network/encryption/Network
...
 		ARG 0 key
+			COMMENT decryption key (e.g. AES secret key or RSA private key)
 		ARG 1 data
 	METHOD m_nmohpzrk encodeRsaPublicKey (Ljava/security/PublicKey;)Ljava/lang/String;
-		ARG 0 publicKey
+		COMMENT Encodes an RSA public {@code key} to a PEM-formatted key string.
+		COMMENT
+		COMMENT <p>
+		COMMENT A PEM-formatted RSA public key is {@value #RSA_PUBLIC_KEY_HEADER}, followed
+		COMMENT by Base64 encoded X.509 encoded key, followed by {@value #RSA_PUBLIC_KEY_FOOTER}.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping]:
```suggestion
		COMMENT by Base64 encoded X.509-encoded key, followed by {@value #RSA_PUBLIC_KEY_FOOTER}.
```
consistency

File: mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping [hunk 4]
```diff
@@ -15,53 +32,158 @@ CLASS net/minecraft/unmapped/C_mgrgevhd net/minecraft/network/encryption/Network
...
+	METHOD m_uznsqkdw generateSecretKey ()Ljavax/crypto/SecretKey;
+		COMMENT {@return a new {@value #SYMMETRIC_BITS} bit AES secret key}
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when generation fails
 	METHOD m_vamsmykn decodeRsaPublicKeyPem (Ljava/lang/String;)Ljava/security/PublicKey;
+		COMMENT Decodes a PEM-formatted RSA public key.
+		COMMENT
+		COMMENT <p>
+		COMMENT A PEM-formatted RSA public key is {@value #RSA_PUBLIC_KEY_HEADER}, followed
+		COMMENT by Base64 encoded X.509 encoded key, followed by {@value #RSA_PUBLIC_KEY_FOOTER}.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping]:
```suggestion
		COMMENT by Base64 encoded X.509-encoded key, followed by {@value #RSA_PUBLIC_KEY_FOOTER}.
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]
Inconsistent with `<thing>-encoded` vs. `<thing> encoded` but otherwise looks good.

**Review by supersaiyansubtlety — APPROVED** [review]

**Pull request merged**